x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/DeleteRoleRequest.java

```diff
@@ -19,27 +20,28 @@
 /**
  * A request delete a role from the security index
  */
-public class DeleteRoleRequest extends ActionRequest implements WriteRequest<DeleteRoleRequest> {
+public class DeleteRoleRequest extends ActionRequest {
 
     private String name;
-    private RefreshPolicy refreshPolicy = RefreshPolicy.IMMEDIATE;
+    private WriteRequest.RefreshPolicy refreshPolicy = WriteRequest.RefreshPolicy.IMMEDIATE;
 
-    public DeleteRoleRequest(StreamInput in) throws IOException {
-        super(in);
-        name = in.readString();
-        refreshPolicy = RefreshPolicy.readFrom(in);
-    }
+    private boolean restrictRequest;
 
     public DeleteRoleRequest() {}
 
-    @Override
-    public DeleteRoleRequest setRefreshPolicy(RefreshPolicy refreshPolicy) {
+    public DeleteRoleRequest setRefreshPolicy(@Nullable String refreshPolicy) {
```

Comment:
This is part of making the action local part of which is removing `implements WriteRequest`